tests/gentropy/common/test_spark_helpers.py

```diff
@@ -0,0 +1,133 @@
+        ("a", 12),
+        ("a", 12),
+        # First value bigger:
+        ("b", 12),
+        ("b", 1),
+        # Second value bigger:
+        ("c", 1),
+        ("c", 12),
+        # First value is null:
+        ("d", None),
+        ("d", 12),
+        # Second value is null:
+        ("e", 12),
+        ("e", None),
+        # Both values are null:
+        ("f", None),
+        ("f", None),
+    ]
+
+    processed_data = (
+        spark.createDataFrame(data, ["group", "value"])
+        .groupBy("group")
+        .agg(
+            f.collect_list(f.struct(f.col("value").alias("value"))).alias("values"),
+            f.max(f.col("value")).alias("max_value"),
+        )
+        .withColumn("sorted_values", order_array_of_structs_by_field("values", "value"))
+        .withColumn("sorted_max", f.col("sorted_values")[0].getField("value"))
+        .select("max_value", "sorted_max")
+        .collect()
+    )
+
+    for row in processed_data:
+        assert row["max_value"] == row["sorted_max"]
+
+
+class TestEnforceSchema:
+    """Test enforce schema."""
+
+    EXPECTED_SCHEMA = t.StructType(
+        [
+            t.StructField("field1", t.StringType(), True),
+            t.StructField("field2", t.StringType(), True),
+            t.StructField("field3", t.StringType(), True),
+        ]
+    )
+
+    @staticmethod
+    @enforce_schema(expected_schema=EXPECTED_SCHEMA)
+    def test_function_good_schema() -> Column:
+        """Create a struct with the expected schema."""
+        return f.struct(
+            f.lit("test1").alias("field1"),
+            f.lit("test2").alias("field2"),
+            f.lit("test3").alias("field3"),
+        )
+
+    @staticmethod
+    @enforce_schema(expected_schema=EXPECTED_SCHEMA)
+    def test_function_missing_column() -> Column:
+        """Create a struct with a missing column."""
+        return f.struct(
+            f.lit("test1").alias("field1"),
+            f.lit("test3").alias("field3"),
+        )
+
+    @staticmethod
+    @enforce_schema(expected_schema=EXPECTED_SCHEMA)
+    def test_function_wrong_order() -> Column:
+        """Create a struct with the wrong order."""
+        return f.struct(
+            f.lit("test2").alias("field2"),
+            f.lit("test1").alias("field1"),
+        )
+
+    @staticmethod
+    @enforce_schema(expected_schema=EXPECTED_SCHEMA)
+    def test_function_extra_column() -> Column:
+        """Create a struct with an extra column."""
+        return f.struct(
+            f.lit("test2").alias("field2"),
+            f.lit("test1").alias("field1"),
+            f.lit("test4").alias("field4"),
+        )
+
+    @staticmethod
+    @enforce_schema(expected_schema=EXPECTED_SCHEMA)
+    def test_function_wrong_type() -> Column:
+        """Create a struct with the wrong type."""
+        return f.struct(
+            f.lit("test2").alias("field2"),
+            f.lit("test1").alias("field1"),
+            f.lit(5).cast(t.IntegerType()).alias("field3"),
```

Comment:
I wonder what happens when it tries to enforce schema in case of opposite (StringType enforced to IntegerType)

Reply:
I assume it is a standard spark type casting, yielding nulls in cases where it is not possible to find a sensible conversion. I have added an extra test case for this. The function certainly won't fail.

Reply:
However I cannot test all possible type conversion (eg. Int to bool)

Reply:
That is fine, I just do not want us to use this function as a black box.